frontend/src/component/pages/shared/CreateShortLinkSection.tsx

```diff
@@ -7,77 +7,188 @@ import { ShortLinkUsage } from './ShortLinkUsage';
 import { Section } from '../../ui/Section';
 import { Url } from '../../../entity/Url';
 import { UIFactory } from '../../UIFactory';
+import { validateLongLinkFormat } from '../../../validators/LongLink.validator';
+import { validateCustomAliasFormat } from '../../../validators/CustomAlias.validator';
+import { raiseCreateShortLinkError } from '../../../state/actions';
+import { IAppState } from '../../../state/reducers';
+import { Store } from 'redux';
+import { UrlService } from '../../../service/Url.service';
+import { QrCodeService } from '../../../service/QrCode.service';
 
-interface Props {
+interface IProps {
+  store: Store<IAppState>;
   uiFactory: UIFactory;
-  longLinkText?: string;
+  urlService: UrlService;
+  qrCodeService: QrCodeService;
+  onShortLinkCreated?: (shortLink: string) => void;
+  onRequestSignIn?: () => void;
+}
+
+interface IState {
+  longLink: string;
   alias?: string;
-  shortLink?: string;
-  inputErr?: string;
-  createdUrl?: Url;
-  qrCodeUrl?: string;
+  inputError: string | null;
```

Comment:
nit: 
TypeScript team doesn't use null : [TypeScript coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined).
Douglas Crockford thinks [null is a bad idea](https://www.youtube.com/watch?v=PSGEjv3Tqo0&feature=youtu.be&t=9m21s)

Reply:
Done.